lib/spack/spack/environment.py

```diff
@@ -492,10 +494,57 @@ def from_dict(base_path, d):
                               d.get('exclude', []),
                               d.get('link', default_view_link))
 
-    def view(self):
-        root = self.root
-        if not os.path.isabs(root):
-            root = os.path.normpath(os.path.join(self.base, self.root))
+    @property
+    def _current_root(self):
+        if not os.path.exists(self.root):
+            return None
+
+        root = os.readlink(self.root)
+        if os.path.isabs(root):
+            return root
+
+        root_dir = os.path.dirname(self.root)
+        return os.path.join(root_dir, root)
+
+    def _next_root(self, specs):
+        content_hash = self.content_hash(specs)
+        root_dir = os.path.dirname(self.root)
+        root_name = os.path.basename(self.root)
+        return os.path.join(root_dir, '._%s_%s' % (root_name, content_hash))
```

Comment:
Is there a reason not to put this in a subdirectory next to the symlink?  If we keep multiple of these around, seems like this will get messy.

Reply:
I don't think it makes much difference in terms of messiness, but I don't see a reason that would be a problem